core/src/consensus/tendermint/worker.rs

```diff
@@ -707,6 +726,47 @@ impl Worker {
             }
             Step::Commit => {
                 cinfo!(ENGINE, "move_to_step: Commit.");
+                let view = state.committed_view().expect("commit always has committed_view");
+                let block_hash = state.committed_block_hash().expect("commit always has committed_block_hash");
+                self.save_last_confirmed_view(view);
+
+                let proposal_received = self.is_proposal_received(self.height, view, block_hash);
+                let proposal_imported = self.client().block(&block_hash.into()).is_some();
+                let best_block_header = self.client().best_block_header();
+                if best_block_header.number() > self.height {
+                    cwarn!(
+                        ENGINE,
+                        "Best block's number {} is larger than tendermint height {} in Commit state",
+                        best_block_header.number(),
+                        self.height
+                    );
+                    return
+                } else if best_block_header.number() == self.height {
+                    cwarn!(
+                        ENGINE,
+                        "Best block's number {} is the same as the tendermint height {} in Commit state",
+                        best_block_header.number(),
+                        self.height
+                    );
+                    return
+                }
+                let best_block_changed = best_block_header.hash() == block_hash;
```

Comment:
```suggestion
let should_update_best_block = best_block_header.hash() != block_hash;
```
this would be more easy to understand.